app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -69,32 +69,27 @@ def perform_auto_provision original_host, rule
   end
 
   def perform_reboot_all hosts = Host::Discovered.all
-    result = true
-    error_message = _("Errors during reboot: %s")
-    overall_errors = ""
+    overall_errors = []
 
     if hosts.count > 0
       hosts.each do |discovered_host|
         begin
           unless discovered_host.reboot
-            error =  "#{discovered_host.name}: failed to reboot "
-            overall_errors << error
-            logger.error error
+            msg = "#{discovered_host.name}: Failed to reboot "
+            e = Excpetion.New()
+            overall_errors << WrappedException.new(e, msg)
+            logger.error e.message
           end
         rescue Exception => e
-          error = "#{discovered_host.name}: #{e.to_s} "
-          overall_errors << error
-          logger.error error
+          overall_errors << e
+          logger.error e.message
         end
       end
     else
-      error_message = _("No discovered hosts to reboot")
-      result = false
-    end
-
-    if overall_errors.present? || !result
-      error_message % overall_errors.strip
+      msg = _("No discovered hosts to reboot")
```

Comment:
It is important to send messages *untranslated* into Foreman::Exception, see https://projects.theforeman.org/projects/foreman/wiki/Translating. Hint: `_()` vs `N_()`.